### ISSUE TITLE
Intermittently blocks site which is being viewed

### ISSUE THREAD

**Issue by dsheroh** [EFForg/privacybadger#1154]
Thus far, I've only seen this on Facebook, but the site occasionally stops working because, while viewing https://www.facebook.com/, Badger decides to completely block (red slider) www.facebook.com.  Reloading the page restores access.  I assume that it is possible for the same to also happen with other sites.

Shouldn't there be a global "never block the site which the primary content is being viewed from" rule to prevent this sort of thing?

**Comment by cooperq** [comment]
Privacy badger should never block the first party domain, so this might be a bug. Another potential thing that could be happening is that facebook loads a frame which has a different origin and that frame in turn loads facebook, which makes facebook look like a third party to privacy badger. I will have to look into this further.

**Comment by dsheroh** [comment]
As I expected, I've run across a second site where this has happened...  Google+.

I think you're likely to at least be on the right track with your frame theory.  On both sites, I've always noticed the problem when an AJAX call fails, either to load the next section of the infinitely-scrolling page or because I've attempted to "like" or add a comment to something.  Of course, that's not entirely conclusive, since the problem could have occurred earlier and I just didn't notice until I attempted to interact with the server...

In any case, though, the initial page load has always worked in my experience, then later interactions with the server fail.
